File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/walkthrough--adding-controls-to-a-worksheet-at-run-time-in-vsto-add-in-project_5.vb
    Sub Application_WorkbookBeforeSave _
        (ByVal workbook As Microsoft.Office.Interop.Excel.Workbook, _
         ByVal SaveAsUI As Boolean, ByRef Cancel As Boolean) _
         Handles Application.WorkbookBeforeSave

        Dim worksheet As Excel.Worksheet = workbook.Worksheets(1)

        If Globals.Factory.HasVstoObject(worksheet) And
            Globals.Factory.GetVstoObject(worksheet).Controls.Count > 0 Then
            Dim vstoWorksheet As Worksheet = Globals.Factory.GetVstoObject(worksheet)

            While vstoWorksheet.Controls.Count > 0
                Dim vstoControl As Object = vstoWorksheet.Controls(0)
                vstoWorksheet.Controls.Remove(vstoControl)
            End While
        End If
    End Sub

File: OLSource1/v1/VS_officedev/codesnippet/CSharp/walkthrough--adding-controls-to-a-worksheet-at-run-time-in-vsto-add-in-project_5.cs
        void Application_WorkbookBeforeSave(Microsoft.Office.Interop.Excel.Workbook workbook, 
            bool SaveAsUI, ref bool Cancel)
        {
            Excel.Worksheet worksheet =
                workbook.Worksheets[1] as Excel.Worksheet;
            
            if (Globals.Factory.HasVstoObject(worksheet) && 
                Globals.Factory.GetVstoObject(worksheet).Controls.Count > 0)
            {
                Worksheet vstoWorksheet = Globals.Factory.GetVstoObject(worksheet);
            
           
                
                while (vstoWorksheet.Controls.Count > 0)
                {
                    object vstoControl = vstoWorksheet.Controls[0];
                    vstoWorksheet.Controls.Remove(vstoControl);
                }
 
            }
        }